var/spack/repos/builtin/packages/python/package.py

```diff
@@ -719,10 +719,13 @@ def config_vars(self):
 for plat_specific in [True, False]:
     config['python_inc'][plat_specific] = get_python_inc(plat_specific, prefix='')
     config['python_lib'][plat_specific] = {}
+    p = {}
     for standard_lib in [True, False]:
-        config['python_lib'][plat_specific][standard_lib] = get_python_lib(
-            plat_specific, standard_lib, prefix=''
-        )
+        p[standard_lib] = get_python_lib(plat_specific, standard_lib, prefix='')
+    # json of python2 does not convert the True/False keys to lowercase, so we have to:
+    config['python_lib'][plat_specific] = dict((str(k).lower(),v) for k,v in p.items())
+config['python_inc'] = dict((str(k).lower(),v) for k,v in config['python_inc'].items())
+config['python_lib'] = dict((str(k).lower(),v) for k,v in config['python_lib'].items())
```

Comment:
I would instead do:
```python
for plat_specific in [True, False]:
    plat_key = str(plat_specific).lower()                                                      
    config['python_inc'][plat_key] = get_python_inc(plat_specific, prefix='')       
    config['python_lib'][plat_key] = {}                                             
    for standard_lib in [True, False]:      
        lib_key = str(standard_lib).lower()                                             
        config['python_lib'][plat_key][lib_key] = get_python_lib(              
            plat_specific, standard_lib, prefix=''                                       
        )
```
We're already looping over this dict once, no need to loop over it again.

Reply:
Right, much better! Updated the commit, ready for review.